api/features_api.py

```diff
@@ -55,8 +55,20 @@ def do_search(self) -> dict[str, Any]:
 
     user_query = self.request.args.get('q', '')
     sort_spec = self.request.args.get('sort')
+    try:
+      start = int(self.request.args.get('start', 0))
+      num = int(self.request.args.get(
+          'num', search.DEFAULT_RESULTS_PER_PAGE))
+    except ValueError:
+      self.abort(400, msg='Invalid start or num')
+    if start < 0:
+      self.abort(400, msg='Parameter out of range: start')
+    if num < 0:
```

Comment:
```suggestion
    if num <= 0:
```
Make it at least 1?

Reply:
I considered that, but it turns out that requesting with num=0 is potentially useful because the response includes the total number of results.